File: lua/Cpp/LuaParserBase.cpp
/* eslint-disable no-underscore-dangle */
/* cspell: ignore antlr, longlong, ULONGLONG, MAXDB */

#pragma once

#include "antlr4-runtime.h"
#include "LuaParserBase.h"
#include "LuaParser.h"

LuaParserBase::LuaParserBase(antlr4::TokenStream* input) : Parser(input)
{
}

bool LuaParserBase::IsFunctionCall()
{
    antlr4::BufferedTokenStream* stream = static_cast<antlr4::BufferedTokenStream*>(_input);
    auto la = stream->LT(1);
    if (la->getType() != LuaParser::NAME) return false;
    la = stream->LT(2);
    if (la->getType() == LuaParser::OP) return false;
    return true;
}

File: lua/JavaScript/LuaParserBase.js
import { Parser } from "antlr4";
import LuaParser from './LuaParser.js';
export default class LuaParserBase extends Parser {
    constructor(input) {
        super(input);
    }
    IsFunctionCall() {
        const stream = this._input;
        var la = stream.LT(1);
        if (la.type !== LuaParser.NAME)
            return false;
        la = stream.LT(2);
        if (la.type === LuaParser.OP)
            return false;
        return true;
    }
}
